src/server/game/Time/UpdateTime.cpp

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (C) 2008-2017 TrinityCore <http://www.trinitycore.org/>
+
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "UpdateTime.h"
+#include "Config.h"
+#include "Log.h"
+
+// create instance
+WorldUpdateTime sWorldUpdateTime;
+
+UpdateTime::UpdateTime()
+{
+    _averageUpdateTime = 0;
+    _totalUpdateTime = 0;
+    _updateTimeTableIndex = 0;
+    _maxUpdateTime = 0;
+    _maxUpdateTimeOfLastTable = 0;
+    _maxUpdateTimeOfCurrentTable = 0;
+
+    _updateTimeDataTable.resize(AVG_DIFF_COUNT, 0);
```

Comment:
hmm completely forgot about it, used vector because i wanted to avoid use of pure table